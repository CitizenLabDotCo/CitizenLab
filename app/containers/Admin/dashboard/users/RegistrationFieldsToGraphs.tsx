// libraries
import React, { PureComponent } from 'react';
import { map } from 'lodash-es';

// resources
import GetUserCustomFields, {
  GetUserCustomFieldsChildProps,
} from 'modules/user_custom_fields/resources/GetUserCustomFields';
import { isNilOrError } from 'utils/helperUtils';

// services
import {
  usersByRegFieldStream,
  IUsersByRegistrationField,
  usersByRegFieldXlsxEndpoint,
} from 'services/stats';

// intl
import { injectIntl } from 'utils/cl-intl';
import { InjectedIntlProps } from 'react-intl';
import localize, { InjectedLocalized } from 'utils/localize';
import messages from '../messages';

// components
import BarChartByCategory from './charts/BarChartByCategory';
import PieChartByCategory from './charts/PieChartByCategory';
import AreaChart from './charts/AreaChart';
import GenderChart from './charts/GenderChart';
import AgeChart from './charts/AgeChart';

interface InputProps {
  currentGroupFilter: string | undefined;
  currentGroupFilterLabel: string | undefined;
  startAt: string | null | undefined;
  endAt: string | null;
}

interface DataProps {
  customFields: GetUserCustomFieldsChildProps;
}

export interface Props extends InputProps, DataProps {}

export class RegistrationFieldsToGraphs extends PureComponent<
  Props & InjectedIntlProps & InjectedLocalized
> {
  convertToGraphFormat = (data: IUsersByRegistrationField) => {
    const {
      series: { users },
      options,
    } = data;
    const res = map(options, (value, key) => {
      return {
        value: users[key] || 0,
        name: this.props.localize(value.title_multiloc),
        code: key,
      };
    });

    if (users['_blank']) {
      res.push({
        value: users['_blank'],
        name: this.props.intl.formatMessage(messages._blank),
        code: '_blank',
      });
    }

    return res.length > 0 ? res : null;
  };
  convertCheckboxToGraphFormat = (data: IUsersByRegistrationField) => {
    const {
      series: { users },
    } = data;
    const res = ['_blank', 'true', 'false'].map((key) => ({
      value: users[key] || 0,
      name: this.props.intl.formatMessage(messages[key]),
      code: 'key',
    }));

    return res.length > 0 ? res : null;
  };

  render() {
    const {
      customFields,
      localize,
      startAt,
      endAt,
      currentGroupFilter,
      currentGroupFilterLabel,
    } = this.props;

    if (isNilOrError(customFields)) {
      return null;
    }

    return customFields.map((field, index) => {
      if (field.attributes.enabled) {
        if (field.attributes.code === 'birthyear') {
          return (
            <AgeChart
              key={index}
              startAt={startAt}
              endAt={endAt}
              currentGroupFilter={currentGroupFilter}
              currentGroupFilterLabel={currentGroupFilterLabel}
            />
          );
        }
        if (field.attributes.input_type === 'number') {
          return;
        }
        if (field.attributes.code === 'gender') {
          return (
            <GenderChart
              key={index}
              startAt={startAt}
              endAt={endAt}
              currentGroupFilter={currentGroupFilter}
              currentGroupFilterLabel={currentGroupFilterLabel}
            />
          );
        }

        if (field.attributes.code === 'domicile') {
          return (
            <AreaChart
              key={index}
              startAt={startAt}
              endAt={endAt}
              currentGroupFilter={currentGroupFilter}
              currentGroupFilterLabel={currentGroupFilterLabel}
            />
          );
        }

        if (field.attributes.input_type === 'checkbox') {
          return (
            <PieChartByCategory
              key={index}
              startAt={startAt}
              endAt={endAt}
              currentGroupFilter={currentGroupFilter}
              currentGroupFilterLabel={currentGroupFilterLabel}
              convertToGraphFormat={this.convertCheckboxToGraphFormat}
              graphTitleString={localize(field.attributes.title_multiloc)}
              stream={usersByRegFieldStream}
              graphUnit="users"
              customId={field.id}
              xlsxEndpoint={usersByRegFieldXlsxEndpoint(field.id)}
            />
          );
        } else {
          return (
            <BarChartByCategory
              key={index}
              startAt={startAt}
              endAt={endAt}
              currentGroupFilter={currentGroupFilter}
              currentGroupFilterLabel={currentGroupFilterLabel}
              convertToGraphFormat={this.convertToGraphFormat}
              graphTitleString={localize(field.attributes.title_multiloc)}
              stream={usersByRegFieldStream}
              graphUnit="users"
              customId={field.id}
              xlsxEndpoint={usersByRegFieldXlsxEndpoint(field.id)}
            />
          );
        }
      }

      return null;
    });
  }
}

const RegistrationFieldsToGraphsWithHoCs = localize<Props>(
  injectIntl<Props & InjectedLocalized>(RegistrationFieldsToGraphs as any)
) as any;

export default (inputProps: InputProps) => (
  <GetUserCustomFields
    inputTypes={['select', 'multiselect', 'checkbox', 'number']}
  >
    {(customFields) => (
      <RegistrationFieldsToGraphsWithHoCs
        {...inputProps}
        customFields={customFields}
      />
    )}
  </GetUserCustomFields>
);
