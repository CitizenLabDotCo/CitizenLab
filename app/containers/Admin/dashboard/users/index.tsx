// libraries
import React, { PureComponent } from 'react';
import moment from 'moment';
import { ThemeProvider } from 'styled-components';

// components
import {
  chartTheme,
  Container,
  GraphsContainer,
  Line,
  GraphCard,
  GraphCardInner,
  GraphCardTitle,
  ControlBar
} from '../';
import TimeControl from '../components/TimeControl';
import IntervalControl from '../components/IntervalControl';
import GenderChart from '../components/GenderChart';
import AgeChart from '../components/AgeChart';

// i18n
import { FormattedMessage } from 'utils/cl-intl';
import messages from '../messages';

interface State {
  interval: 'weeks' | 'months' | 'years';
  intervalIndex: number;
  currentGroupFilter?: string;
}

type Props = {

};

export default class UsersDashboard extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      interval: 'months',
      intervalIndex: 0,
      currentGroupFilter: undefined
    };
  }

  changeInterval = (interval: 'weeks' | 'months' | 'years') => {
    this.setState({ interval, intervalIndex: 0 });
  }

  changeIntervalIndex = (intervalIndex: number) => {
    this.setState({ intervalIndex });
  }

  handleOnGroupFilter = (filter) => {
    // To be implemented
    this.setState({ currentGroupFilter: filter });
  }

  render() {
    const { interval, intervalIndex, currentGroupFilter } = this.state;
    const startAtMoment = moment().startOf(interval).add(intervalIndex, interval);
    const endAtMoment = moment(startAtMoment).add(1, interval);
    const startAt = startAtMoment.toISOString();
    const endAt = endAtMoment.toISOString();

    return (
      <Container>
        <ControlBar>
          <TimeControl
            value={intervalIndex}
            interval={interval}
            onChange={this.changeIntervalIndex}
            currentTime={startAtMoment}
          />
          <IntervalControl
            value={interval}
            onChange={this.changeInterval}
          />
        </ControlBar>
        <ThemeProvider theme={chartTheme}>
          <GraphsContainer>
            <Line>
              <GraphCard className="first halfWidth">
                <GraphCardInner>
                  <GraphCardTitle>
                    <FormattedMessage {...messages.usersByGenderTitle} />
                  </GraphCardTitle>
                  <GenderChart startAt={startAt} endAt={endAt} currentGroupFilter={currentGroupFilter} />
                </GraphCardInner>
              </GraphCard>
              <GraphCard className="halfWidth">
                <GraphCardInner>
                  <GraphCardTitle>
                    <FormattedMessage {...messages.usersByAgeTitle} />
                  </GraphCardTitle>
                  <AgeChart startAt={startAt} endAt={endAt} currentGroupFilter={currentGroupFilter} />
                </GraphCardInner>
              </GraphCard>
            </Line>
          </GraphsContainer>
        </ThemeProvider>
      </Container>
    );
  }
}
