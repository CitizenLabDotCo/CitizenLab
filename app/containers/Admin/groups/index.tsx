// Libraries
import React from 'react';

// Components
import HelmetIntl from 'components/HelmetIntl';

// i18n
import messages from './messages';

export default class GroupsDashboard extends React.PureComponent {
  render() {
    return (
      <div id="e2e-groups-admin-dashboard">
        <HelmetIntl
          title={messages.helmetTitle}
          description={messages.helmetDescription}
        />
        {this.props.children}
      </div>
    );
  }
}
