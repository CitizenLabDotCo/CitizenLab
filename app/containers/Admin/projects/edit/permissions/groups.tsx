// Libraries
import * as React from 'react';
import * as Rx from 'rxjs/Rx';
import * as _ from 'lodash';

// i18n
import { injectIntl, InjectedIntlProps, FormattedMessage, FormattedHTMLMessage } from 'react-intl';
import { getLocalized } from 'utils/i18n';
import messages from './messages';

// Components
import Button from 'components/UI/Button';
import Icon from 'components/UI/Icon';
import GroupAvatar from 'containers/Admin/groups/all/GroupAvatar';

// Services
import { groupsProjectsByProjectIdStream, IGroupsProjects } from 'services/groupsProjects';

// Style
import styled from 'styled-components';

const EmptyStateMessage = styled.p`
  background: rgba(1, 161, 177, 0.07);
  border-radius: 5px;
  color: #01A1B1;
  display: flex;
  align-items: center;
  font-size: 1.15rem;
  padding: 1.5rem;

  background: color: ${props => props.theme.colorMain || '#333'};
`;

const StyledIcon = styled(Icon)`
  height: 1em;
  margin-right: 2rem;
`;

const Container = styled.div`
  width: 100%;
`;

const ListWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  margin-top: -2rem;
`;

const ListItem = styled.div`
  align-items: center;
  border-bottom: 1px solid #EAEAEA;
  display: flex;
  justify-content: space-between;

  > * {
    margin: 2rem 1rem;

    &:first-child {
      margin-left: 0;
    }

    &:last-child {
      margin-right: 0;
    }
  }

  > .expand {
    flex: 1;
  }
`;

// Typing
interface Props {
  projectId: string;
}

interface State {
  groupsProjects: IGroupsProjects | null;
}

class ProjectGroups extends React.PureComponent<Props & InjectedIntlProps, State> {
  state: State;
  subscriptions: Rx.Subscription[];

  constructor() {
    super();
    this.state = {
      groupsProjects: null
    };
    this.subscriptions = [];
  }

  componentWillMount() {
    const { projectId } = this.props;
    const groupsProjects$ = groupsProjectsByProjectIdStream(projectId).observable;

    this.subscriptions = [
      groupsProjects$.subscribe((groupsProjects) => {
        console.log('groupsProjects:');
        console.log(groupsProjects);
        this.setState({ groupsProjects });
      })
    ];
  }

  componentWillUnmount() {
    this.subscriptions.forEach(subscription => subscription.unsubscribe());
  }

  /*
  const ListWrapper = styled.div`
    display: flex;
    flex-direction: column;
    width: 100%;
    margin-top: -2rem;
  `;

  const ListItem = styled.div`
    align-items: center;
    border-bottom: 1px solid #EAEAEA;
    display: flex;
    justify-content: space-between;

    > * {
      margin: 2rem 1rem;

      &:first-child {
        margin-left: 0;
      }

      &:last-child {
        margin-right: 0;
      }
    }

    > .expand {
      flex: 1;
    }
  `;

  <ListWrapper className="e2e-groups-list">
    {groups.map((group) => (
      <ListItem key={group.id}>
        <GroupAvatar groupId={group.id} />
        <p className="expand">
          {getLocalized(group.attributes.title_multiloc, locale, tenantLocales)}
        </p>
        <p className="expand">
          <FormattedMessage {...messages.members} values={{ count: group.attributes.memberships_count }} />
        </p>
        <Button onClick={this.createDeleteGroupHandler(group.id)} style="text" circularCorners={false} icon="delete">
          <FormattedMessage {...messages.deleteButtonLabel} />
        </Button>
        <Button linkTo={`/admin/groups/edit/${group.id}`} style="secondary" circularCorners={false} icon="edit">
          <FormattedMessage {...messages.editButtonLabel} />
        </Button>
      </ListItem>
    ))}
  </ListWrapper>
  */

  render() {
    const { formatMessage } = this.props.intl;
    const { groupsProjects } = this.state;

    const noGroups = (!groupsProjects ? (
      <EmptyStateMessage>
        <StyledIcon name="warning" />
        <FormattedHTMLMessage {...messages.noSelectedGroupsMessage} />
      </EmptyStateMessage>
    ) : null);

    const groupsList = (!groupsProjects ? (
      <span>GroupsList</span>
    ) : null);

    return (
      <Container>
        {noGroups}
        {groupsList}
      </Container>
    );
  }
}

export default injectIntl<Props>(ProjectGroups);
