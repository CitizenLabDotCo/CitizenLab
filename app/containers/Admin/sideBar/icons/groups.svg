<svg xmlns="http://www.w3.org/2000/svg" width="22" height="20" viewBox="0 0 22 20">
    <path fill="#FFF" fill-rule="nonzero" d="M6.703 14.613a.57.57 0 0 1-.794 0 6.903 6.903 0 0 1-1.668-2.659.53.53 0 0 1 .11-.536.572.572 0 0 1 .538-.171c.198.04.357.181.418.368a5.82 5.82 0 0 0 1.405 2.24c.21.213.206.55-.009.758zm8.98.154a.568.568 0 0 0 .402-.163 6.908 6.908 0 0 0 1.668-2.658.53.53 0 0 0-.114-.533.573.573 0 0 0-.533-.17.556.556 0 0 0-.42.361 5.822 5.822 0 0 1-1.405 2.24.53.53 0 0 0-.116.59.562.562 0 0 0 .518.333zM9.653 4.141a6.254 6.254 0 0 1 2.694 0 .573.573 0 0 0 .55-.15.531.531 0 0 0 .129-.54.555.555 0 0 0-.428-.368 7.426 7.426 0 0 0-3.199 0 .555.555 0 0 0-.417.37.531.531 0 0 0 .128.534c.139.143.346.202.543.154zm5.491-1.571C15.185 1.228 16.32.16 17.707.16c1.386 0 2.521 1.068 2.562 2.41-.04 1.341-1.176 2.409-2.562 2.409-1.387 0-2.522-1.068-2.563-2.41zm1.123 0c.03.748.666 1.34 1.44 1.34.773 0 1.41-.592 1.44-1.34-.03-.75-.667-1.341-1.44-1.341-.774 0-1.41.592-1.44 1.34zm5.612 4.68v1.493a1.014 1.014 0 0 1-.315.728c-.2.192-.472.3-.754.299h-6.218a1.082 1.082 0 0 1-.754-.3 1.014 1.014 0 0 1-.315-.727V7.25c.009-1.147.975-2.071 2.16-2.065h.577a.57.57 0 0 1 .393.156l1.05 1 1.05-1a.57.57 0 0 1 .394-.156h.577a2.185 2.185 0 0 1 1.523.598c.406.388.637.915.64 1.467h-.008zm-1.123 0c-.008-.547-.472-.984-1.037-.978h-.339l-1.107 1.054a.832.832 0 0 1-1.135 0L16.03 6.272h-.348c-.565-.006-1.029.431-1.037.978v1.433h6.12l-.01-1.433zM1.731 2.57C1.77 1.228 2.907.16 4.293.16c1.387 0 2.522 1.068 2.563 2.41C6.815 3.91 5.68 4.979 4.293 4.979c-1.386 0-2.521-1.068-2.562-2.41zm1.122 0c.03.748.667 1.34 1.44 1.34.774 0 1.41-.592 1.44-1.34-.03-.75-.666-1.341-1.44-1.341-.773 0-1.41.592-1.44 1.34zM.113 8.743V7.25c.008-1.147.974-2.071 2.159-2.065h.579a.57.57 0 0 1 .393.156l1.05 1 1.051-1a.57.57 0 0 1 .393-.156h.577c1.185-.006 2.15.918 2.16 2.065v1.493c-.003.569-.48 1.028-1.067 1.027H1.181a1.082 1.082 0 0 1-.754-.3 1.014 1.014 0 0 1-.315-.727zm1.122-.06h6.12V7.25c-.01-.547-.473-.984-1.038-.978H5.97L4.862 7.326a.832.832 0 0 1-1.135 0L2.62 6.272h-.348c-.565-.006-1.029.431-1.037.978v1.433zm12.329 3.997c.013 1.007-.603 1.923-1.56 2.317A2.626 2.626 0 0 1 9.2 14.48a2.426 2.426 0 0 1-.57-2.71c.394-.931 1.331-1.54 2.371-1.54 1.406-.008 2.554 1.089 2.564 2.45zm-1.123 0a1.394 1.394 0 0 0-.866-1.312 1.476 1.476 0 0 0-1.582.281c-.419.397-.548 1-.327 1.525.22.526.748.87 1.334.87.786.008 1.431-.602 1.441-1.364zm2.741 4.68v1.494a1.014 1.014 0 0 1-.315.728c-.2.192-.471.3-.753.298H7.886a1.082 1.082 0 0 1-.753-.298 1.014 1.014 0 0 1-.315-.728v-1.493c.008-1.148.976-2.073 2.162-2.065h.577a.57.57 0 0 1 .392.156l1.051 1 1.05-1a.57.57 0 0 1 .393-.156h.577c1.186-.008 2.154.917 2.162 2.065zm-1.122 0c-.009-.547-.474-.985-1.04-.977h-.345l-1.107 1.052a.832.832 0 0 1-1.134 0L9.328 16.38H8.98c-.565-.006-1.029.432-1.038.979v1.432h6.118v-1.43z" opacity=".6"/>
</svg>
