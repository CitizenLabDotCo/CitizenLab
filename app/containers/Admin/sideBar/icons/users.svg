<svg xmlns="http://www.w3.org/2000/svg" width="21" height="22" viewBox="0 0 21 22">
    <g fill="none" fillRule="evenodd" stroke="#FFF" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" opacity=".6" transform="translate(1 1)">
        <circle cx="9.091" cy="10" r="9.091"/>
        <circle cx="9.091" cy="8.977" r="2.537"/>
        <path d="M9.09 12.198a5.074 5.074 0 0 0-5.065 5.346 9.09 9.09 0 0 0 5.066 1.547 9.09 9.09 0 0 0 5.06-1.548c.007-.09.012-.18.014-.27a5.074 5.074 0 0 0-5.074-5.075z"/>
    </g>
</svg>
