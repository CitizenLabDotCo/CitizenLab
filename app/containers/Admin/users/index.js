import React from 'react';
import PropTypes from 'prop-types';
import HelmetIntl from 'components/HelmetIntl';

// components
// import { FormattedMessage } from 'react-intl';
// import WatchSagas from 'utils/containers/watchSagas';
// import sagas from 'resources/users/sagas';

// store
// import { preprocess } from 'utils';

// messages
import messages from './messages';

class UserDashboard extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function

  render() {
    return (
      <div>
        <HelmetIntl
          title={messages.helmetTitle}
          description={messages.helmetDescription}
        />
        {this.props.children}
      </div>

    );
  }
}


UserDashboard.propTypes = {
  children: PropTypes.element,
};

export default UserDashboard;
