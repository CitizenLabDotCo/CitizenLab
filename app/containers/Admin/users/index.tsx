import React, { SFC } from 'react';

// components
import HelmetIntl from 'components/HelmetIntl';
import TabbedResource from 'components/admin/TabbedResource';

// resources
import GetLocation, { GetLocationChildProps } from 'resources/GetLocation';

// i18n
import messages from './messages';
import { InjectedIntlProps } from 'react-intl';
import { injectIntl } from 'utils/cl-intl';

interface InputProps {}

interface DataProps {
  location: GetLocationChildProps;
}

interface Props extends InputProps, DataProps {}

const UsersPage: SFC<Props & InjectedIntlProps> = ({ location, children, intl }) => {
  const { formatMessage } = intl;

  const tabs = [
    { label: formatMessage(messages.tabRegisteredUsers), url: '/admin/users/registered' },
    { label: formatMessage(messages.tabInviteByEmail), url: '/admin/users/invitations' },
  ];

  const resource = {
    title: formatMessage(messages.viewPublicResource)
  };

  if (!location) return null;

  return (
    <>
      <TabbedResource
        resource={resource}
        messages={messages}
        tabs={tabs}
        location={location}
      >
        <HelmetIntl
          title={messages.helmetTitle}
          description={messages.helmetDescription}
        />
        {children}
      </TabbedResource>
    </>
  );
};

const UsersPageWithIntl = injectIntl(UsersPage);

export default (inputProps: InputProps) => (
  <GetLocation>
    {location => <UsersPageWithIntl {...inputProps} location={location} />}
  </GetLocation>
);
