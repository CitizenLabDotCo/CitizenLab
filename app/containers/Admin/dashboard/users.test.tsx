// libraries
import React from 'react';
import moment from 'moment';

// component to test
// ! it is the component without the HoCs (data, localization...)
import { UsersDashboard } from './users';

// mock data and functions to replace the HoCs
import { mockGetGroups } from 'services/__mocks__/groups';
import { localizeProps } from 'utils/testUtils/localizeProps';
import { shallowWithIntl } from 'utils/testUtils/withIntl';

// what needs to be mocked by jest to render the component
jest.mock('utils/cl-intl');

describe('<UsersDashboard />', () => {
  it('renders correctly', () => {
    // passing in mock data to replace the HoCs
    // ignore the error, shallowWithIntl method passes in the intl object
    const wrapper = shallowWithIntl(<UsersDashboard groups={mockGetGroups} {...localizeProps} />);
    // the component sets endAtMoment to current time, but to match snapshots we need a stable date
    wrapper.setState({ endAtMoment: moment('2010-01-01T05:06:07', moment.ISO_8601) });
    expect(wrapper).toMatchSnapshot();
  });

  it('reacts to group changes', () => {
    const wrapper = shallowWithIntl(<UsersDashboard groups={{ ...mockGetGroups, groupsList: null }} {...localizeProps} />);
    let filters = wrapper.find('ChartFilters');
    // setting snapshots here is a disputable option -- cf code discussion
    expect(filters.prop('groupFilterOptions')).toMatchSnapshot();

    wrapper.setProps({ groups: mockGetGroups });

    // you can access the instance and read its props and state like this
    // don't abuse it -- cf code discussion
    // console.log(wrapper.instance().props);

    // you need to update the filter object to reflect the changes
    filters = wrapper.find('ChartFilters');
    expect(filters.prop('groupFilterOptions')).toMatchSnapshot()
  });
});
