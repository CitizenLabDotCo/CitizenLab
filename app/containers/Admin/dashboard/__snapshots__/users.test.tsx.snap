// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersDashboard /> reacts to group changes 1`] = `
Array [
  Object {
    "label": "All Groups",
    "value": "",
  },
]
`;

exports[`<UsersDashboard /> reacts to group changes 2`] = `
Array [
  Object {
    "label": "All Groups",
    "value": "",
  },
  Object {
    "label": "Test Group",
    "value": "test_group_id",
  },
]
`;

exports[`<UsersDashboard /> renders correctly 1`] = `
<Fragment>
  <dashboard__ControlBar>
    <TimeControl
      endAtMoment={"2010-01-01T04:06:07.000Z"}
      onChange={[Function]}
    />
  </dashboard__ControlBar>
  <ChartFilters
    configuration={
      Object {
        "showGroupFilter": true,
        "showProjectFilter": false,
        "showTopicFilter": false,
      }
    }
    groupFilterOptions={
      Array [
        Object {
          "label": "All Groups",
          "value": "",
        },
        Object {
          "label": "Test Group",
          "value": "test_group_id",
        },
      ]
    }
    onGroupFilter={[Function]}
    onProjectFilter={null}
    onTopicFilter={null}
    projectFilterOptions={null}
    topicFilterOptions={null}
  />
  <ThemeProvider
    theme={[Function]}
  >
    <dashboard__GraphsContainer>
      <AgeChart
        endAt="2010-01-01T04:06:07.000Z"
      />
      <Component
        endAt="2010-01-01T04:06:07.000Z"
      />
      <WithTheme(MostActiveUsersList)
        className="dynamicHeight"
        endAt="2010-01-01T04:06:07.000Z"
        infoMessage="A user gets 5 points per posted idea, 3 points per posted comment and 1 point per vote."
      />
    </dashboard__GraphsContainer>
  </ThemeProvider>
</Fragment>
`;
